src/response.rs

```diff
@@ -522,7 +519,7 @@ impl<'a, 'b> Response<'a, 'b> {
         let mime = path
             .extension()
             .and_then(|ext| to_mime(&ext.to_string_lossy()))
-            .unwrap_or(Mime(TopLevel::Application, SubLevel::Ext("octet-stream".into()), vec![]));
+            .unwrap_or_else(|| Mime(TopLevel::Application, SubLevel::Ext("octet-stream".into()), vec![]));
```

Comment:
I'm not sure about this, since the `"octet-stream"` string will be converted and allocated even if the value isn't used. The question is which alternative is the cheapest.